scripts/tasks/download-translations.js

```diff
@@ -0,0 +1,170 @@
+//@ts-check
+const stream = require('stream');
+const { join } = require('path');
+const { promisify } = require('util');
+const pipeline = promisify(stream.pipeline);
+const got = require('got').default;
+const unzipper = require('unzipper');
+const { tmpdir } = require('os');
+const fs = require('fs-extra');
+
+// Assumes running from the root of the repo
+const OUTPUT_PATH = join(process.cwd(), 'temp-i18n');
+const { CROWDIN_TOKEN, CROWDIN_PROJECT_ID } = process.env;
+const PROJECT_ID = parseInt(CROWDIN_PROJECT_ID);
+const crowdin = require('@crowdin/crowdin-api-client').default;
+
+const { translationsApi } = new crowdin({
+  token: CROWDIN_TOKEN,
+});
+
+/**
+ * Downloads the Crowdin file and unzips the contents
+ * @param {string} url
+ * @param {string} destination
+ */
+const downloadFiles = async (url, destination) => {
+  const tmpPath = join(
+    tmpdir(),
+    'electronjs.org',
+    `${Math.ceil(Math.random() * 1000)}`
+  );
+
+  await pipeline(got.stream(url), unzipper.Extract({ path: tmpPath }));
+
+  const contentPath = join(tmpPath, `[electron.i18n] master`, `content`);
+
+  const locales = await fs.readdir(contentPath);
+
+  for (const locale of locales) {
+    const localeDestination = join(destination, locale.toLowerCase());
+    const docsDestination = join(
+      localeDestination,
+      'docusaurus-plugin-content-docs',
+      'current',
+      'latest'
+    );
+    await fs.mkdirp(docsDestination);
+
+    await fs.copy(join(contentPath, locale, 'docs'), docsDestination);
+
+    await fs.copy(
+      join(contentPath, locale, 'website', 'i18n'),
+
+      localeDestination
+    );
+  }
+
+  return locales.map((locale) => locale.toLowerCase());
+};
+
+/**
+ * Waits for the given number of seconds
+ * @param {number} seconds
+ * @returns
+ */
+const waitFor = (seconds) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, seconds * 1000);
+  });
+};
+
+/**
+ * @param {number} buildId
+ * @returns
+ */
+const getBuild = async (buildId) => {
+  const builds = await translationsApi.listProjectBuilds(PROJECT_ID);
+  const build = builds.data.find((item) => item.data.id === buildId);
+
+  return build.data;
+};
+
+/**
+ * Kicks a build for `PROJECT_ID` and returns the download link once it
+ * has finished.
+ * If a build is not kicked we risked downloading outdated files.
+ * @param {number} projectId
+ */
+const buildAndDownloadLink = async (projectId) => {
+  const {
+    data: { id: buildId },
+  } = await translationsApi.buildProject(projectId);
+
+  let counter = 10;
+  let interval = 30;
+  let build;
+
+  for (let i = 0; i < counter; i++) {
+    build = await getBuild(buildId);
+    console.log(`Crowdin status: Project ${buildId} - ${build.status}`);
+
+    if (build.status === 'finished') {
+      break;
+    } else {
+      console.log(
+        `Crowdin status: Waiting ${interval} seconds (retry ${i}/${counter})`
```

Comment:
It doesn't look like there are many concurrent builds so this isn't a huge deal but it looks like https://support.crowdin.com/api/v2/#operation/api.projects.translations.builds.get might be more efficient / convenient than listProjectBuilds, since the former targets a single build id instead of returning all the builds.

For that matter, looks like you could just keep calling builds/{buildId}/download until it stops giving  you a 202. The nice thing is that payload returns both the status string for the console.log on line 102, but also a progress % that would be nice to have in the log too

Reply:
Great suggestion!